### PR TITLE
Implemented methods in StreamHomework class

### PR DESCRIPTION
**Pull request by Yuri13579**

### CODE REVIEW

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 1]
```diff
@@ -6,55 +6,99 @@
 
 public class Range<T> implements Set<T> {
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: You have to implement Set (which contains unique elements).

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 2]
```diff
@@ -6,55 +6,99 @@
...
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
+
+        return (T1[]) arrayList.toArray();
     }
 
     public boolean add(T t) {
-        return false;
+        return  arrayList.add(t);
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java:39]: You have to check if the element is unique.

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 3]
```diff
@@ -6,55 +6,99 @@
...
+            for (float j = v; j <= v1; j += 0.1f) {
+                range.add(j);
+            }
+        }
+        return range;
+
+
+    }
+
+    public static Range<Character> of(char a, char d, Function<Character, Character> function) {
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java:83]: Need to support custom types

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 4]
```diff
@@ -6,55 +6,99 @@
...
+               range.add(j);
+           }
+       }
+        return range;
+    }
+
+    public static Range<Float> of(float v, float v1) {
+
+        Range<Float> range= new Range<>();
+        if (v<v1) {
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java]: Looks like method **of** above, try not to duplicate code

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 2 above)
Comment (Yuri13579) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java:39]: public boolean add(T t) {
        if (arrayList.contains(t)) return false;
        return  arrayList.add(t);
    }

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 4 above)
Comment (Yuri13579) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: public static Range<Float> of(float v, float v1) {

        return of(v, v1, f -> (f + 0.1f));

    }

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 3 above)
Comment (Yuri13579) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java:83]: public static <T extends Comparable> Range<T> of (T e1, T e2, Function<T, T> function) {
        Range<T> range = new Range<>();
        if (e1.compareTo(e2) < 0) {
            T i = e1;
            while (i.compareTo(e2) <= 0) {
                range.add(i);
                i = function.apply(i);
            }
        }
        return range;
    }

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 2 above)
Comment (george-stotle) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java:39]: Ok. Update the Pull Request please

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 4 above)
Comment (george-stotle) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: Yes, update PR.

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 3 above)
Comment (george-stotle) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java:83]: Update PR, why do you check with if?

Comment (george-stotle) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java:83]: And how do you use generic for static method? Run tests

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 5]
```diff
@@ -1,60 +1,105 @@
...
+        return of(i, i1, integer -> (integer + 1));
+    }
+
+    public static Range<Float> of(float v, float v1) {
+
+        return of(v, v1, f -> (f + 0.1f));
+
+    }
+
+    public static Range<Character> of(char a, char d, Function<Character, Character> function) {
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java:83]: Isn't it covered by method **of** below?

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 6]
```diff
@@ -1,60 +1,105 @@
...
     public Object[] toArray() {
-        return new Object[0];
+
+        return arrayList.toArray();
     }
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
+
+        return (T1[]) arrayList.toArray();
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java:34]: pass a here

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 2 above)
Comment (Yuri13579) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java:39]: already updated

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java (hunk 1 above)
Comment (Yuri13579) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: What does it mean?

Comment (Yuri13579) [reply src/main/java/co/inventorsoft/academy/collections/model/Range.java]: that is to be done in this way?                                                                                                     public class Range<T>  implements Set<T>  {

    private T arrayElement[];
    int size;

File: src/main/java/co/inventorsoft/academy/collections/model/Range.java [hunk 7]
```diff
@@ -1,60 +1,105 @@
...
 
     public boolean containsAll(Collection<?> c) {
-        return false;
+        return arrayList.containsAll(c);
     }
 
     public boolean addAll(Collection<? extends T> c) {
-        return false;
+
+        return arrayList.addAll(c);
```
Comment (george-stotle) [inline src/main/java/co/inventorsoft/academy/collections/model/Range.java:53]: you haven't checked for duplication here

**Pull request closed without merging**